pkg/ocm/managedservice.go

```diff
@@ -66,7 +76,8 @@ func (c *Client) CreateManagedService(args CreateManagedServiceArgs) (*msv1.Mana
 								WorkerRoleARN(args.AwsWorkerRoleARN)).
 							OperatorIAMRoles(operatorIamRoles...)).
 						AccountID(args.AwsAccountID).
-						SubnetIDs(args.SubnetIDs...)).
+						SubnetIDs(args.SubnetIDs...).
+						PrivateLink(args.Privatelink)).
```

Comment:
Why are we passing both `ListeningMethod` and `PrivateLink`? Shouldn't this logic be done in the MS? (i.e. if PrivateLink is set, set the ListeningMethod accordingly, or vice versa)

Reply:
We do this today to ensure that the interactive flow during `rosa create cluster` is consistent. The API should do its own validation as well to ensure that you can't have public privatelink clusters.

Reply:
>  The API should do its own validation as well to ensure that you can't have public privatelink clusters.

I tested this out, and once the request gets to CS, cluster creation fails if the listening method isn't set to "Internal".

So are we ok with keeping this logic in Rosa, or should it be moved to MS?

Reply:
Yes, I'm okay with keeping this in ROSA for now

Reply:
I won't block on this, but I'm still not sure I understand why we need to pass two api parameters to MS here instead of one and letting MS set the other one accordingly. 
The MS api doesn't have to match the CS api. We can set the cluster listening mode inside MS according to the privatelink parameter instead of passing it from ROSA. 
Or do we want to expose the listening mode as another option for the customers? I much prefer to keep the MS api as minimal as possible.